### PR TITLE
Generate the code of search option only when searchEngine option is present

### PR DESCRIPTION
**Pull request by maznag**
Generate the code of search option only when searchEngine option is present.

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by murdos** [comment]
Could your provide more information about the issue you're fixing?

**Comment by maznag** [comment]
@murdos  The related code of search option is generated even if the searchEngine option isn't present. due to this test condition: `searchEngine !== false` whitch present in several files (for example in entity-management.component.ts.ejs). In fact, when searchEngine option isn't present this test condition always evaluate to `true` because searchEngine hase '' as value. 

To reproduce the issue:
* Generate a new app with the default options without specifying searchEngine option
* Generate a sample entity
* Check the generated code and you will see that the related code of search option will be generated.

File: generators/entity/index.js [hunk 1]
```diff
@@ -162,7 +162,7 @@ class EntityGenerator extends BaseBlueprintGenerator {
                 context.prodDatabaseType = configuration.get('prodDatabaseType') || this.options.db;
                 context.devDatabaseType = configuration.get('devDatabaseType') || this.options.db;
                 context.skipFakeData = configuration.get('skipFakeData');
-                context.searchEngine = configuration.get('searchEngine');
+                context.searchEngine = _.isUndefined(configuration.get('searchEngine')) ? false : configuration.get('searchEngine');
```
Comment (mshima) [inline generators/entity/index.js:165]: I prefer this notation:
```suggestion
                context.searchEngine = _configuration.get('searchEngine')) || false;
```

Comment (maznag) [reply generators/entity/index.js]: done !

**Comment by DanielFran** [comment]
Thanks @maznag

**Pull request merged**